presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -265,10 +266,16 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
     {
         Optional<Table> table = metastore.getTable(tableName.getSchemaName(), tableName.getTableName());
-        if (!table.isPresent() || table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+        if (!table.isPresent()) {
             throw new TableNotFoundException(tableName);
         }
 
+        if (table.get().getTableType().equals(TableType.VIRTUAL_VIEW.name())) {
+            Optional<String> comment = Optional.ofNullable(table.get().getParameters().get(TABLE_COMMENT));
+
+            return new ConnectorTableMetadata(tableName, new ArrayList<>(), emptyMap(), comment);
```

Comment:
Use `ImmutableList.of()` instead of `new ArrayList<>()`. And use `ImmutableMap.of()` instead of `emptyMap()`. In general, use immutable collections if it's not expected to be modified.